apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V62__new_schema_snf_claims.sql

```diff
@@ -0,0 +1,843 @@
+-- NEW_SCHEMA_SNF_CLAIMS.SQL
+-- flyway migration for SNF_CLAIMS and SNF_CLAIM_LINES tables into
+-- a new schema structure that:
+--   1) changes data type of CLM_ID, CLM_GROUP_ID, BENE_ID to BIGINT.
+--   2) organizes parent claim table (SNF_CLAIMS) such that common
+--      claims data is organized at top of column definition.
+--
+-- Once current table data is migrated to new table name/strucutre,
+-- updatable db views are created wrapping the new table structure;
+-- this allows BFD services entity beans to temporarily use updatable
+-- data view(s) instead of actual database table names. This strategy
+-- facilitates and promotes minimal app/database downtime during that
+-- period when table structure and data are being migrated and app
+-- services code is deployed.
+--
+-- HSQL differs from PSQL (postgres) in that the table defintion
+-- must be explicitly declared prior to loading data into the
+-- target table. PSQL can derive the table structure based on
+-- the data input (i.e., column name, data type). Thus, for HSQL,
+-- we need to explicitly define the table structure prior to loading data.
+--
+-- For HSQL, explicitly define/create a new SNF_CLAIMS_NEW table in
+-- the current PUBLIC schema
+${logic.hsql-only}	create table public.snf_claims_new (
+	${logic.hsql-only}	clm_id								bigint NOT NULL,
+	${logic.hsql-only}	bene_id								bigint NOT NULL,
+	${logic.hsql-only}	clm_grp_id							bigint NOT NULL,
```

Comment:
I didn't realize we were doing clm_grp_id now. We don't have the pre-work done for the application for that column. Is that going to make the corresponding application change too complex?

Reply:
So CLM_GRP_ID will have zero minimal impact; it is defined as a numeric right now. The same holds true for things like LINE_NUM...currently defined as NUMERIC but is actually a SMALLINT (SHORT) and should be defined in db as such.

So the pre-work that you did is not impacted at all; in the Java code, we are passing around numeric data type for CLM_GRP_ID.

The original 'charter' of the schema work was to make things as small as possible, leading to storage savings and performance gains. So CLM_GRP_ID is currently defined as a numeric. However, If precision is not required, you should not use the NUMERIC type because calculations on NUMERIC values are typically slower than [integers](https://www.postgresqltutorial.com/postgresql-integer/), floats, and double precisions.